cmd/kube-apiserver/app/server.go

```diff
@@ -673,19 +664,21 @@ func buildServiceResolver(enabledAggregatorRouting bool, hostname string, inform
 }
 
 func getServiceIPAndRanges(serviceClusterIPRanges string) (net.IP, net.IPNet, net.IPNet, error) {
-	serviceClusterIPRangeList := []string{}
+	var serviceClusterIPRangeList []string
 	if serviceClusterIPRanges != "" {
 		serviceClusterIPRangeList = strings.Split(serviceClusterIPRanges, ",")
 	}
 
-	var apiServerServiceIP net.IP
-	var primaryServiceIPRange net.IPNet
-	var secondaryServiceIPRange net.IPNet
-	var err error
+	var (
+		apiServerServiceIP      net.IP
+		primaryServiceIPRange   net.IPNet
+		secondaryServiceIPRange net.IPNet
+	)
+
 	// nothing provided by user, use default range (only applies to the Primary)
 	if len(serviceClusterIPRangeList) == 0 {
 		var primaryServiceClusterCIDR net.IPNet
-		primaryServiceIPRange, apiServerServiceIP, err = controlplane.ServiceIPRange(primaryServiceClusterCIDR)
+		primaryServiceIPRange, apiServerServiceIP, err := controlplane.ServiceIPRange(primaryServiceClusterCIDR)
```

Comment:
if we're going to intentionally shadow the variables from the parent scope, I'd suggest moving the variable declarations below the `if len(serviceClusterIPRangeList) == 0 {` block to make it clear those are not related to this assignment

Reply:
`ServiceIPRange` expects a `net.IPNet` whereas `ParseCIDRSloppy` returns a `*net.IPNet` this is why I kept it as it was. I can revert back the whole if you prefer